### PR TITLE
[refactor](multi catalog)Refactor FileQueryScanNode init and finalize mothods.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Refactor FileQueryScanNode init and finalize methods.
Handle schema related initialization in init method, handle scan range generation in finalize method.

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/planner/FileLoadScanNode.java [hunk 1]
```diff
@@ -214,54 +199,14 @@ public void finalize(Analyzer analyzer) throws UserException {
         for (int i = 0; i < contexts.size(); ++i) {
             FileLoadScanNode.ParamCreateContext context = contexts.get(i);
             FileScanProviderIf scanProvider = scanProviders.get(i);
-            setDefaultValueExprs(scanProvider, context);
+            setDefaultValueExprs(scanProvider, context.srcSlotDescByName, context.params);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/FileLoadScanNode.java:202]: I think this can be moved to `init`

**Review by morningman — APPROVED** [review]
LGTM

**Review by morningman — CHANGES_REQUESTED** [review]
Need to modify

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**